packages/flutter_test/lib/src/controller.dart

```diff
@@ -23,6 +23,144 @@ const double kDragSlopDefault = 20.0;
 
 const String _defaultPlatform = kIsWeb ? 'web' : 'android';
 
+/// Class that programatically interacts with the Semantics tree.
+class SemanticsController {
+  /// Creates a SemanticsController that uses the given binding.
+  SemanticsController(this.binding);
```

Comment:
Are users ever expected to create their own, or should this be private?

Reply:
For now, I went with protecting the constructor. My gut instinct is that people will need to be able to override this, so I've made it easy for people to extend this and pass their custom implementations to `WidgetController`, but not easy to instantiate `SemanticsController` directly.

Reply:
With the additional feedback from @chunhtai, I ended up making this constructor private to prevent the `SemanticController` from being instantiated outside of this library. We can make this more public and easier to replace at a later date if need be.